pkg/istio-agent/agent.go

```diff
@@ -450,26 +451,32 @@ func (a *Agent) initSdsServer() error {
 // It will not return until both workload cert and root cert are generated.
 //
 // TODO: evaluate replacing the STS server with a file data source, to simplify Envoy config
+// TODO: Fix this method with unused return value
+// nolint: unparam
 func (a *Agent) getWorkloadCerts(st *cache.SecretManagerClient) (sk *security.SecretItem, err error) {
-	b := backoff.NewExponentialBackOff()
-	b.MaxElapsedTime = 0
-
-	for {
+	b := backoff.NewExponentialBackOff(func(off *backoff.ExponentialBackOff) {
+		off.MaxElapsedTime = 0
```

Comment:
does this mean we may eventually backoff for days?

Reply:
Yes, i found it is the original meaning. I am going to fix it.